### PR TITLE
Tcvp 138 setup test automation

### PR DESCRIPTION
**Pull request by harshjaingov**
# Description

This PR includes the following proposed change(s):

TCVP-138: SetUp Test Automation for Traffic Courts

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by rantifamilusi — APPROVED** [review]

File: .github/workflows/cucumber-tests.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  cucumber-tests:
+    name: Runs cucumber-tests
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (harshjaingov) [reply .github/workflows/cucumber-tests.yml:13]: done

File: .github/workflows/cucumber-tests.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    steps:
+      - name: Pull Git repo.
```
[the remark this answers is not in the document]
Comment (harshjaingov) [reply .github/workflows/cucumber-tests.yml:15]: done

File: .github/workflows/cucumber-tests.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Run Frontend UI
+        run: |
+          cd src/frontend/citizen-portal
+          sudo npm install
```
[the remark this answers is not in the document]
Comment (harshjaingov) [reply .github/workflows/cucumber-tests.yml:34]: done

File: tests/src/test/java/stepDefinitions/frontendstepdefinitions/Steps.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                  .release(canvasElement)
+                   .build();
+	    signature.perform();
+	    
+	    sleep();
+		
+	    driver.findElement(By.xpath("//input[@type='button' and @value='Complete']")).click();
+	}
+	@Then("The Violation Ticket should be Successfully Submitted")
+	public void the_violation_ticket_should_be_successfully_submitted() {
```
[the remark this answers is not in the document]
Comment (harshjaingov) [reply tests/src/test/java/stepDefinitions/frontendstepdefinitions/Steps.java:98]: done. added test to the step

File: tests/src/test/resources/frontend/frontendTest.feature [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Feature: User initiates and submits Dispute in BC Traffic Courts Online Website
+
+Scenario:  User initiates Dispute in BC Traffic Courts Online Website
```
[the remark this answers is not in the document]
Comment (harshjaingov) [reply tests/src/test/resources/frontend/frontendTest.feature]: done

File: tests/src/test/java/stepDefinitions/frontendstepdefinitions/Steps.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (harshjaingov) [reply tests/src/test/java/stepDefinitions/frontendstepdefinitions/Steps.java:98]: done. implemented tests

File: tests/src/test/resources/frontend/frontendTest.feature [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+	Given User has successfully launched the web browser
+	When User navigates to BC Traffic Courts Online Website
+	And User clicks on Initiate Dispute Option
+	And User Enters the Violation Ticket Details
+	And User Signs and Clicks on Complete Option
+	Then The Violation Ticket should be Successfully Submitted
```
[the remark this answers is not in the document]
Comment (harshjaingov) [reply tests/src/test/resources/frontend/frontendTest.feature]: done

File: tests/pom.xml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
+                    <encoding>UTF-8</encoding>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
[the remark this answers is not in the document]
Comment (harshjaingov) [reply tests/pom.xml]: done

File: .github/workflows/cucumber-tests.yml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        run: |
+          cd ${{github.workspace}}/src/backend/TrafficCourtsApi
+          docker build -f Dockerfile -t disputev1 .
+          docker run -p 5004:80 disputev1 &
+
+      - name: Running UI and API Tests
+        run: |
+
+          # Maven requires chrome driver.
+          wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
```
[the remark this answers is not in the document]
Comment (harshjaingov) [reply .github/workflows/cucumber-tests.yml]: done. implemented the changes

File: tests/src/test/java/stepDefinitions/frontendstepdefinitions/Steps.java [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+	}
+	@When("User Signs and Clicks on Complete Option")
+	public void user_signs_and_clicks_on_complete_option() {
+	    // Write code here that turns the phrase above into concrete actions
+		
+		WebElement canvasElement = driver.findElement(By.xpath("//canvas[@tabindex='0']"));
+
+	    Actions builder = new Actions(driver);
+	    Action signature = builder.contextClick(canvasElement) //start points x axis and y axis. 
+	              .clickAndHold()
```
[the remark this answers is not in the document]
Comment (harshjaingov) [reply tests/src/test/java/stepDefinitions/frontendstepdefinitions/Steps.java]: done

**Pull request merged**